app/Providers/AuthServiceProvider.php

```diff
@@ -25,6 +27,23 @@ public function boot()
     {
         $this->registerPolicies();
 
-        //
+        //Authorisations
+
+        // When a registration is requested
+        Gate::define('view-registration', function (User $user, Registration $registration) {
```

Comment:
Do these need to be different? Or could we just have one gate 'registration-action' ?

Reply:
I'm not entirely sure where Gates are going, so i split 'em all out into individual actions.

Reply:
But you can use the same rule - and put in different places.

So unless some users can view but not update, or print but not view, you only need one gate defined.

If the gate is different, it should let a different set of users in.